### ISSUE TITLE
Setup of zha is taking over 10 seconds. -- homeassistant

### ISSUE THREAD

**Issue by kenthua** [zigpy/bellows#120]
I see plenty of these references on the home assistant forums and other forums as well, however there is never really a clear way to debug other than, is your zigbee.db in the absolute path, reboot a few times and it started working.  I'm not sure if my device is just dead or there is some configuration conflict causing problems.

Is my device broken?  I only got it working once since I received it, paired 1 switch and then it stopped starting, even with a new zigbee.db

It seems nothing happens after this statement, as evident in bellows CLI itself

```
2018-05-19 16:07:37 DEBUG (MainThread) [bellows.uart] Sending: b'1ac038bc7e'
```

homeassistant.log
```
...
2018-05-19 16:07:33 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=group>
2018-05-19 16:07:33 INFO (MainThread) [homeassistant.loader] Loaded tts from homeassistant.components.tts
2018-05-19 16:07:33 INFO (MainThread) [homeassistant.loader] Loaded tts.google from homeassistant.components.tts.google
2018-05-19 16:07:33 INFO (MainThread) [homeassistant.loader] Loaded zha from homeassistant.components.zha
...
2018-05-19 16:07:37 INFO (MainThread) [homeassistant.setup] Setting up zha
2018-05-19 16:07:37 DEBUG (MainThread) [zigpy.appdb] Loading application state from /config/zigbee.db
2018-05-19 16:07:37 DEBUG (MainThread) [bellows.uart] Sending: b'1ac038bc7e'
2018-05-19 16:07:37 INFO (MainThread) [homeassistant.setup] Setting up updater
2018-05-19 16:07:37 INFO (MainThread) [homeassistant.setup] Setup of domain updater took 0.0 seconds.
2018-05-19 16:07:37 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=updater>
2018-05-19 16:07:38 INFO (MainThread) [homeassistant.setup] Setting up discovery
2018-05-19 16:07:38 INFO (MainThread) [homeassistant.setup] Setup of domain discovery took 0.0 seconds.
2018-05-19 16:07:38 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=discovery>
2018-05-19 16:07:38 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.yr
...
2018-05-19 16:07:40 INFO (MainThread) [homeassistant.setup] Setup of domain notify took 6.4 seconds.
2018-05-19 16:07:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=notify>
2018-05-19 16:07:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=config.config_entries>
2018-05-19 16:07:40 INFO (MainThread) [homeassistant.setup] Setup of domain config took 6.7 seconds.
2018-05-19 16:07:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=config>
2018-05-19 16:07:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: domain=tts, service=google_say>
2018-05-19 16:07:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_registered[L]: domain=tts, service=clear_cache>
2018-05-19 16:07:41 INFO (MainThread) [homeassistant.setup] Setup of domain tts took 1.8 seconds.
2018-05-19 16:07:41 INFO (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=tts>
2018-05-19 16:07:47 WARNING (MainThread) [homeassistant.setup] Setup of zha is taking over 10 seconds.
```

dmesg 
- most say this is a red herring?
```
[   28.021620] hassio: port 3(vethdc89de9) entered blocking state
[   28.021622] hassio: port 3(vethdc89de9) entered forwarding state
[   34.032822] udevd[7]: starting version 3.2.4
[   34.050590] udevd[7]: starting eudev-3.2.4
[   35.934614] cp210x ttyUSB1: failed set req 0x1e size 4 status: -32
[   35.934626] cp210x ttyUSB1: failed to set baud rate to 300
```

belllows
```
bash-4.4# bellows -v DEBUG -d /dev/ttyUSB1 info
debug: Using selector: EpollSelector
debug: Connected. Resetting.
debug: Sending: b'1ac038bc7e'
^C
Aborted!
bash-4.4# bellows -v DEBUG -d /dev/ttyUSB1 -b 115200 info
debug: Using selector: EpollSelector
debug: Connected. Resetting.
debug: Sending: b'1ac038bc7e'
^C
Aborted!
bash-4.4# bellows -v DEBUG -d /dev/ttyUSB1 -b 57600 info
debug: Using selector: EpollSelector
debug: Connected. Resetting.
debug: Sending: b'1ac038bc7e'
^C
Aborted!
```

https://github.com/home-assistant/home-assistant/issues/8922#issuecomment-389666865

**Comment by kenthua** [comment]
Looks like the zigpy or bellows inside the hassio docker image needs to be updated or configured differently.  On the same device if I run the same command in a different container, things look much better.

```
$ sudo docker run -it --device /dev/ttyUSB1 python bash

root@948e5ed58421:/# bellows -v DEBUG -d /dev/ttyUSB1 info
debug: Using selector: EpollSelector
debug: Connected. Resetting.
debug: Sending: b'1ac038bc7e'
debug: RSTACK Version: 2 Reason: RESET_SOFTWARE frame: b'c1020b0a527e'
debug: Send command version
debug: Sending: b'004221a850ed2c7e'
debug: Data frame: b'0142a1a8502805e67f627e'
debug: Sending: b'8160597e'
debug: Application frame 0 (version) received
debug: Configuring...
debug: Send command setConfigurationValue
debug: Sending: b'7d314321fb582815c3277e'
debug: Data frame: b'1243a1fb54fb737e'
debug: Sending: b'82503a7e'
debug: Application frame 83 (setConfigurationValue) received
debug: Send command setConfigurationValue
debug: Sending: b'224021fb592f15226f7e'
debug: Data frame: b'2340a1fb54c6107e'
debug: Sending: b'83401b7e'
debug: Application frame 83 (setConfigurationValue) received
debug: Send command setConfigurationValue
debug: Sending: b'334121fb792b15a2d77e'
debug: Data frame: b'3441a1fb54d32a7e'
debug: Sending: b'8430fc7e'
debug: Application frame 83 (setConfigurationValue) received
debug: Send command setConfigurationValue
debug: Sending: b'444621fb55d51534da7e'
debug: Data frame: b'4546a1fb5435d07e'
debug: Sending: b'8520dd7e'
debug: Application frame 83 (setConfigurationValue) received
debug: Send command networkInit
debug: Sending: b'554721bfb6607e'
debug: Data frame: b'5647a5bf54b4247e'
debug: Sending: b'8610be7e'
debug: Application frame 23 (networkInit) received
debug: Data frame: b'6647b1b1c487df7e'
debug: Sending: b'87009f7e'
debug: Application frame 25 (stackStatusHandler) received
debug: Send command getEui64
debug: Sending: b'6744218e08c37e'
debug: Data frame: b'7744a18ea70343b959fb4725ad9b7e'
debug: Sending: b'8070787e'
debug: Application frame 38 (getEui64) received
[00:0d:6f:00:0b:56:29:f3]
debug: Send command getNodeId
debug: Sending: b'7045218f65587e'
debug: Data frame: b'0045a18f542af0d57e'
debug: Sending: b'8160597e'
debug: Application frame 39 (getNodeId) received
[0]
debug: Send command networkState
debug: Sending: b'014a21b0ba147e'
debug: Data frame: b'114aa1b05617a27e'
debug: Sending: b'82503a7e'
debug: Application frame 24 (networkState) received
[<EmberNetworkStatus.JOINED_NETWORK: 2>]
debug: Send command getNetworkParameters
debug: Sending: b'124b21803b0c7e'
debug: Data frame: b'224ba180542b9fa97fa936b3120f78e6944127abedce677302c12ca97e'
debug: Sending: b'83401b7e'
debug: Application frame 40 (getNetworkParameters) received
[<EmberStatus.SUCCESS: 0>, <EmberNodeType.COORDINATOR: 1>, <EmberNetworkParameters extendedPanId=[138, 27, 38, 61, 124, 150, 184, 90] panId=45034 radioTxPower=8 radioChannel=15 joinMethod=EmberJoinMethod.USE_MAC_ASSOCIATION nwkManagerId=0 nwkUpdateId=0 channels=134215680>]
debug: Send command getCurrentSecurityState
debug: Sending: b'234821c160e47e'
debug: Data frame: b'3348a1c1545e154170c24125c5589275bc7e'
debug: Sending: b'8430fc7e'
debug: Application frame 105 (getCurrentSecurityState) received
[<EmberStatus.SUCCESS: 0>, <EmberCurrentSecurityState bitmask=116 trustCenterLongAddress=00:0d:6f:00:0b:56:29:f3>]
```

**Comment by kenthua** [comment]
Yup, you are correct from what I'm seeing as I went through the debugging process.  So it's the version of zigpy / bellows that's installed and/or configured on the hass.io docker image.

**Comment by rcloran** [comment]
Yes, the dependencies were recently updated.

**Comment by kenthua** [comment]
So I updated to the beta version of the docker image `homeassistant/qemux86-64-homeassistant   0.70.0b3`.  I confirmed that zigpy was 0.1.0 and bellows 0.6.0 libraries were installed.

Still having [issues](https://pastebin.com/U6DNLiAZ) with it starting.  On one occasion I was able to jump start it by running the [bellows info](https://pastebin.com/nMdq8A3z) command within the container, not sure if it was a coincidence.  homeassistant was still in a broken state though, didn't fully initialize.

**Comment by dmulcahey** [comment]
please see the comment I added [here](https://github.com/home-assistant/home-assistant/issues/8922#issuecomment-392349100)  and make sure this isn't what is causing the issue.

**Comment by kenthua** [comment]
Not sure if this helps the debugging process, but I was able to re-create this behavior in a standalone python container with zigpy and bellows installed.  

This type of behavior appeared as well when the homeassistant container also tried to control/send commands to ttyUSB1 as well.  

```
root@cfc0cbd40e95:/# bellows -d /dev/ttyUSB1 -v DEBUG info
debug: Using selector: EpollSelector
debug: Connected. Resetting.
debug: Sending: b'1ac038bc7e'
^C
Aborted!
```

When I shut off the homeassistant container, it ACK as expected along with many other request/response messages

@dmulcahey In my `lsusb` checking across multiple reboots, the positioning of the zwave and zigbee radios stay consistent.  Same with my testing of the /dev/ttyUSB1 with a standalone container.

**Comment by pumbarger** [comment]
I too am having this same problem. I switched to a hassbien installation (rp 3b+) because I had better access to things since they are not in docker containers.  I checked the required library levels in the zha component and they are the newer versions given above.
```
REQUIREMENTS = [
    'bellows==0.6.0',
    'zigpy==0.1.0',
    'zigpy-xbee==0.1.0',
]
```
This does not seem to be specific to hass.io unless hassbien is just a variant of it.

**Comment by kenthua** [comment]
Something is up with the x86_64 homeassistant hassio docke image, because I installed HA on a python:alpine image and there were no issues loading zha.

**Comment by coreyo** [comment]
Was this issue closed as a duplicate, or as a docker flaw?  I'm having the same problem in ubuntu 16.04 LTS using virtualenv (no docker).

**Comment by kenthua** [comment]
I closed it as a hassio docker image flaw

@coreyo Are you able to run the bellows commands successfully outside of home assistant?

**Comment by pumbarger** [comment]
This may be a homeassistant issue but it is not unique to x86_64 as I am having the same issue on a raspberry pi 3b with HASS.IO and with Haspien.

I suspect the HUSBZB-1 controller I just purchased may be bad.  Running:
bellows -d /dev/ttyUSB1 -v DEBUG info

Would that be expected to hang if the underlying hardware were bad?  I would expect some kind of timeout and returning bad data or an error.

**Comment by kenthua** [comment]
Is it hanging here?
```
debug: Using selector: EpollSelector
debug: Connected. Resetting.
debug: Sending: b'1ac038bc7e'
```

**Comment by coreyo** [comment]
@kenthua Yes, that's exactly what it's doing.  pulling the device out and plugging it in a few times didn't seem to fix the issue.

@pumbarger While it's always possible there is a large batch of faulty devices, I think you're jumping to the wrong conclusion.  There seem to be dozens of us all with the same problem.  More likely the issue is that the version of bellows or zigpy has an issue.

**Comment by coreyo** [comment]
@kenthua just to be a little more thorough, these commands were issued in my virtualenv:
```
(homeassistant) hass@router:/etc/homeassistant$ groups hass
hass : hass dialout
```
```
(homeassistant) hass@router:/etc/homeassistant$ ps aux |grep hass
hass      1591  0.0  0.6 1064964 51824 ?       Ssl  May27   0:59 node /usr/local/lib/node_modules/smartthings-mqtt-bridge/bin/smartthings-mqtt-bridge
hass     25357  104  0.8 1855396 67892 ?       Ssl  11:19   0:05 /opt/homeassistant/bin/python3.6 /opt/homeassistant/bin/hass -c /etc/homeassistant --log-file /var/log/homeassistant/hass.log
```
```
(homeassistant) hass@router:/etc/homeassistant$ pip list |grep -E 'zigpy|bellows|homeassistant'
bellows                 0.6.0     
homeassistant           0.70.0    
zigpy                   0.1.0     
zigpy-xbee              0.1.0  
```
```
(homeassistant) hass@router:/etc/homeassistant$ ls -l /dev/ttyUSB*
crw-rw---- 1 root dialout 188, 0 May 31 00:24 /dev/ttyUSB0
crw-rw---- 1 root dialout 188, 1 May 31 00:29 /dev/ttyUSB1
```
```
(homeassistant) hass@router:/etc/homeassistant$ bellows -v DEBUG -d /dev/ttyUSB1 infodebug: Using selector: EpollSelector
debug: Connected. Resetting.
debug: Sending: b'1ac038bc7e'
^C
Aborted!
```
I've got a little Celeron N3150 mini PC with 2 gigabit ethernet ports running Ubuntu Xenial LTS as a dedicated router.  Home assistant is running in an virtualenv as a dedicated 'hass' user.  It has both an aeotec zwave dongle (/dev/ttyACM0) and the HUSBZB-1 dongle (/dev/ttyUSB*)

**Comment by pumbarger** [comment]
@kenthua asked: Are you able to run the bellows commands successfully outside of home assistant?
I tried with a haspian distro and it hung just as @coreyo shows.  But in hindsight I am not sure that the HA init was still waiting on the same command to complete.  I should have killed that process first but I am not sure if I did or not.  The info provided by @coreyo looks much more detailed and reliable.

**Comment by kenthua** [comment]
@pumbarger yea, if HA is running and stuck, it will block other tests from working.

**Comment by coreyo** [comment]
ahh, I should have shown that the problem persisted even after I stopped homeassistant.  That is indeed the case.

**Comment by kenthua** [comment]
I take it all back, once I added a device to my zigbee network, same issue.

**Comment by kenthua** [comment]
I got my environment back up and running a in virtualenv.

**Comment by coreyo** [comment]
@kenthua not sure what you mean by that last comment.  Is it working properly, or are you experiencing the same 'hanging' behavior as the rest of us in your virtualenv?

**Comment by kenthua** [comment]
Yes, it works with virtualenv for me

**Comment by coreyo** [comment]
@kenthua mine still hangs.  Even tried restarting the system.  Did you do anything special?  Do you have intermittent problems restarting homeassistant?

**Comment by drbios** [comment]
Hello every one.
i use hassbian 0.70.1 rapsberry pi 3+ and a Xbee s2c conected tu a usb explorer. inicially i had the 10 second problem and web client not loading. but after configuring the xbee unit with  **API MODE ENABLED** ass coordinator  it's working perfect.
maybe some other usb zigbbe units suffers the same problem.
let me know if this help you 
best regards

**Comment by dmulcahey** [comment]
http://www.sysrun.io/2017/10/20/quickfix-zigbeez-wave-usb-on-qnap/

Stumbled across this... I wonder if some of these docker installs don’t have these modules as well...

**Comment by kenthua** [comment]
@dmulcahey This one for the qnap issues seems more if the ttyUSBX devices don't show up at all.  The docker images do have the cpxxxx module loaded.

**Comment by dmulcahey** [comment]
Agreed but There are multiple modules listed. I’m wondering if the serial one could be the culprit...

**Comment by coreyo** [comment]
@dmulcahey I admit I've been wondering if it's a module/kernel issue as well.  I tried the following:

> coreyo@router:~$ sudo modprobe usbserial cp210x cdc-acm
> coreyo@router:~t$ sudo lsmod |grep -E 'usbserial|cp210x|cdc-acm'
> cp210x                 24576  0
> usbserial              53248  1 cp210x

I'm assuming that the usbserial and cdc-acm are compiled directly into the kernel.  At this point, I unplugged my HUSBZB-1 dongle and plugged it back in.  In my virtualenv, I did the following:

> (homeassistant) hass@router:/etc/homeassistant$ bellows -v DEBUG -d /dev/ttyUSB1 info
> debug: Using selector: EpollSelector
> debug: Connected. Resetting.
> debug: Sending: b'1ac038bc7e'
> ^C
> Aborted!

No luck.  I did some more searching and I did find something that might help us to collectively diagnose what is going on.  Try the following command on your own machines:

> coreyo@router:~$ sudo udevadm info -q property -a /dev/ttyUSB1
> 
> Udevadm info starts with the device specified by the devpath and then
> walks up the chain of parent devices. It prints for every device
> found, all possible attributes in the udev rules key format.
> A rule to match, can be composed by the attributes of the device
> and the attributes from one single parent device.
> 
>   looking at device '/devices/pci0000:00/0000:00:14.0/usb1/1-3/1-3:1.1/ttyUSB1/tty/ttyUSB1':
>     KERNEL=="ttyUSB1"
>     SUBSYSTEM=="tty"
>     DRIVER==""
> 
>   looking at parent device '/devices/pci0000:00/0000:00:14.0/usb1/1-3/1-3:1.1/ttyUSB1':
>     KERNELS=="ttyUSB1"
>     SUBSYSTEMS=="usb-serial"
>     DRIVERS=="cp210x"
>     ATTRS{port_number}=="0"
> 
>   looking at parent device '/devices/pci0000:00/0000:00:14.0/usb1/1-3/1-3:1.1':
>     KERNELS=="1-3:1.1"
>     SUBSYSTEMS=="usb"
>     DRIVERS=="cp210x"
>     ATTRS{authorized}=="1"
>     ATTRS{bAlternateSetting}==" 0"
>     ATTRS{bInterfaceClass}=="ff"
>     ATTRS{bInterfaceNumber}=="01"
>     ATTRS{bInterfaceProtocol}=="00"
>     ATTRS{bInterfaceSubClass}=="00"
>     ATTRS{bNumEndpoints}=="02"
>     ATTRS{interface}=="HubZ ZigBee Com Port"
>     ATTRS{supports_autosuspend}=="1"
> 
>     --- snip ---

As the description says, it walks up the device tree in reverse from child to parent.  You can see that the cp210x module is in use for the core device's parent and grandparent.

Incidentally, this is also a great way to identify which node corresponds to your zwave radio and which corresponds to your zigbee radio.  See the following:

> coreyo@router:~$ sudo udevadm info -q property -a /dev/ttyUSB0 |grep "ATTRS{interface}"
>     ATTRS{interface}=="HubZ Z-Wave Com Port"
> coreyo@router:~$ sudo udevadm info -q property -a /dev/ttyUSB1 |grep "ATTRS{interface}"
>     ATTRS{interface}=="HubZ ZigBee Com Port"

**Comment by Yoda-x** [comment]
@kenthua please try to assign the vendor/product id to the cpxxx driver
my zigbee stick shows with lsusb as:

> Bus 001 Device 011: ID 10c4:89fb Cygnal Integrated Products, Inc. 

I did a

> echo 10c4 89fb > /sys/bus/usb-serial/drivers/cp210x/new_id

and plug-off/in and it created a ttyusb device

**Comment by kenthua** [comment]
@Yoda-x I don't have issues with the /dev/ttyUSBX devices showing up.  They are also consistently mapping zwave to ttyUSB0 and zigbee to ttyUSB1.   

It works fine in a virtualenv, but in a docker container something happens to the way the driver/library uses talks to the devices.

**Comment by coreyo** [comment]
@kenthua I too have no problems with the consistency of assignment to ttyUSB0 and ttyUSB1 as zwave and zigbee respectively.

In your first post, you seemed to indicate that you had problems at first, but finally got it working.  Can you tell us anything about how you set it up and what you did to make it work in your virtualenv?  Can you also tell us what build/version/kernel of linux you are using?

**Comment by coreyo** [comment]
@Yoda-x 
```
root@router:~# whoami
root
root@router:~# lsusb |grep Cygnal
Bus 001 Device 008: ID 10c4:8a2a Cygnal Integrated Products, Inc.
root@router:~# echo '10c4:8a2a' > /sys/bus/usb-serial/drivers/cp210x/new_id
bash: echo: write error: Invalid argument
root@router:~# ls -l /sys/bus/usb-serial/drivers/cp210x/new_id
-rw-r--r-- 1 root root 4096 Jun  5 11:42 /sys/bus/usb-serial/drivers/cp210x/new_id
```

**Comment by kenthua** [comment]
@coreyo My initial issues had stemmed from using the hassio qemu x86_64 build image.  I then tried to use the homeassistant image which also caused issues.  I think I tried just a vanilla python alpine image and had the same problem.   So at least for me it's docker container specific.  Which is why I opened this one up, https://github.com/home-assistant/hassio/issues/490

So the next step was to try virtualenv and I was able to get it to work.  Though there were times when I had to restart the homeassistant startup because it had issues.

Setup wise, standard, per homeassistant instructions. https://www.home-assistant.io/docs/installation/virtualenv/

I have issues with autostart via systemd, but it's been functional with manual start.

Fedora 28 - 4.16.11-300.fc28.x86_64

**Comment by coreyo** [comment]
@kenthua okay, that might be an important distinction.  If you log in as your homeassistant user, activate the virtual env via 'source /path/to/virtualenv/bin/activate', then trying out the bellows command?

How are you manually starting homeassistant?

**Comment by kenthua** [comment]
@coreyo 
I can't run the bellows command because I let homeassistant manage the dependencies in the deps folder.  However, since it starts fine, the bellows info command would operate as expected as well.

I just start the hass binary as noted in the instructions.

**Comment by Yoda-x** [comment]
@coreyo 
`root@router:~# echo '10c4:8a2a' > /sys/bus/usb-serial/drivers/cp210x/new_id
`
without the columns -> 

> echo 10c4 8a2a > /sys/bus/usb-serial/drivers/cp210x/new_id

**Comment by corylulu** [comment]
I'm having the same issue with a standalone setup (not Docker) on a RPi3 with a HUSBZB-1 adapter. I tested both with and without a virtual environment setup and got the same results. I also tried rolling back to `homeassistant==0.69.1` and am experiencing the same issue on `bellows==0.5.4`, `zigpy==0.0.3`. I have no clue what I should do.  Installed this all fresh yesterday, so everything was up-to-date.

**Comment by coreyo** [comment]
@kenthua  homeassistant works within the virtual environment same as you might via the commandline.  To try the bellows command as a standalone, simply do the following:

log in as whatever user you're using to run home assistant
`$ sudo su - <homeassistant user>`

cd to the homeassistant directory
`$ cd /path/to/homeassistant/dir`

activate the virtual environment and run bellows:
```
$ source bin/activate
$ bellows -v DEBUG -d /dev/ttyUSB1 inf
```

let me know if it hangs for youu

**Comment by kenthua** [comment]
@coreyo as I mentioned earlier, my bellows and zigpy libraries are installed within the dep folder so it's not within the default python bin / lib path.  However to run the test you wanted, I created a new virtualenv installed bellows and zigpy.  I then ran the bellows command successfully.

**Comment by kenthua** [comment]
@corylulu you need to test directly with the bellows command without homeassistant running, if you can successfully get that running then debug homeassistant, otherwise with just homeassistant you may drive yourself crazy.

**Comment by coreyo** [comment]
@Yoda-x thanks.  I think you meant to say "replace the colon with a space".   I haven't had any problems with the nodes appearing, but thought it was worth a try at this point.  Unfortunately, this doesn't seem to fix anything.

**Comment by Yoda-x** [comment]
yes, I dont think it is something with the docker image or driver support. More with the raspi io performance via USB.
I recently switched from a intel nuc to a raspi3b and now have the same problems:
can you enable debugging for bellows? Want to check if you see also this error.

> 2018-06-06 00:49:34 DEBUG (MainThread) [bellows.uart] Error frame: b'c20251a8bd7e'

in configuration.yaml -> 
<pre>
logger:
  default: error
  logs:
    bellows: debug
</pre>
 
reading the silabs [UG101,](https://www.silabs.com/documents/public/user-guides/UG101.pdf) chapter 6 gives a possible hint.

**Comment by coreyo** [comment]
@Yoda-x This is the same output that I see when I run the bellows command directly.

There may be some issues specific to the Raspberry Pi.  I have seen several reviews of the product where the reviewer complained about flakey zigbee performance via home assistant on the raspberry pi until they moved it to a PC or laptop.

However, I'm seeing the same problems described on this thread on a celeron PC.  I suspect that this behavior is a separate issue.

**Comment by Yoda-x** [comment]
I coded in the bellows uart code the ASH retransmit feature with some debugging for frames that get lost between the zigbee stick and the raspberry usb device. during the start phase I see some frames getting lost, which may result in the zigbee stick going to fail mode, without proper feedback from bellows.
I will let it run for a day and then make a pull request, when my stick is still alive

**Comment by coreyo** [comment]
@Yoda-x great! At the very least it will rule that out as an issue.  If you do find anything, let me know how I can get your code changes and I'll test it out on my celeron box.  I still suspect that there might be an issue with the kernel/modules that could cause the same behavior.

**Comment by Yoda-x** [comment]
Just get this file https://github.com/Yoda-x/bellows/blob/ash_retransmit/bellows/uart.py and copy it over the original file in your ha installation. But keep a copy of the original one :-)

**Comment by kenthua** [comment]
I think we are dealing with separate issues, so it'd be good to isolate.

1. docker: homeassistant - Stuck at `debug: Sending: b'1ac038bc7e'`
2. rpi: homeassistant on virtualenv / bare - `Error frame: b'c20251a8bd7e'`

On my non-rpi, it works fine in virtualenv but doesn't in docker.  I don't recall getting an `Error frame: b'c20251a8bd7e'` error.

**Comment by coreyo** [comment]
@Yoda-x @kenthua I think you're right, that's why I wanted to compare.  After replacing uart.py with Yoda's version, I still see the following behavior:

```
(homeassistant) hass@router:~/lib/python3.6/site-packages/bellows$ bellows -v DEBUG -d /dev/ttyUSB1 info
debug: Using selector: EpollSelector
debug: Connected. Resetting.
debug: Sending: b'1ac038bc7e'
^C
Aborted!
```

It still seems to hang on the first transmit.  Is there more to be done to see the added debugging?

**Comment by Yoda-x** [comment]
@coreyo I see an error in my code. stay tuned

**Comment by corylulu** [comment]
@kenthua Yeah, that's what I was doing. It does hang on me.... I've never gotten it to work. I bought to through the same seller as another guy that said he found out it didn't actually have a zigbee radio or something though. So I'm returning it and trying to get one from a different seller.

**Comment by coreyo** [comment]
@corylulu that's easy enough to verify, see my previous post: 
```
$ sudo udevadm info -q property -a /dev/ttyUSB0 |grep "ATTRS{interface}"
$ sudo udevadm info -q property -a /dev/ttyUSB1 |grep "ATTRS{interface}"
```
If either of those say 'ATTRS{interface}=="HubZ ZigBee Com Port"' then you've got the zwave/zigbee dongle.  Likely, if you have /dev/ttyUSB1 at all, you have the right dongle.

**Comment by Yoda-x** [comment]
@coreyo : please give it a try if you find some time. I can see that at least the raspi3 is a little bit sensitive on the USB port. I have an usb stick for logs and database in one usb port. Accessing the data creates dropped frames for my zigbee stick.

@kenthua :   @coreyo  Sending: b'1ac038bc7e' is the ASH reset frame which initializes the UART connection
>2018-06-06 18:46:38 DEBUG (MainThread) [bellows.uart] Sending: b'1ac038bc7e'
>2018-06-06 18:46:40 DEBUG (MainThread) [bellows.uart] RSTACK Version: 2 Reason: RESET_SOFTWARE frame: b'c1020b0a527e'
See the UG101 doc I linked above. I think this is the first try to write something to the usb port. Do you have the right permissions to write to the ttyUSB port inside docker?

**Comment by kenthua** [comment]
@Yoda-x Yes, the container was started in --privileged, which is not ideal, but it is just to make sure it has what it needs.

Below is the expected output for an ACK of `Sending: b'1ac038bc7e'` , assuming there aren't any additional frames not being show by the debug.
```
debug: RSTACK Version: 2 Reason: RESET_SOFTWARE frame: b'c1020b0a527e'
```

**Comment by coreyo** [comment]
@Yoda-x  still the same behavior with your latest code:
```
(homeassistant) hass@router:~/lib/python3.6/site-packages/bellows$ bellows -v DEBUG -d /dev/ttyUSB1 info
debug: Using selector: EpollSelector
debug: Connected. Resetting.
debug: Sending: b'1ac038bc7e'
^C
Aborted!
```

**Comment by Yoda-x** [comment]
I think there are two issues as @kenthua stated. The one with **b'1ac038bc7e`** with docker, and the dropped frames with **Error frame: b'c20251a8bd7e'**. 
My patch is for **Error frame: b'c20251a8bd7e'**. 
 @coreyo: did you tested from inside docker? 
Looking for the docker issue, I see many forums with users on docker-ce who state that access to ttyUSB is not working, There seems an bug with the privileges where they needed to use "--group-add=dialout" to the docker cmd to make the ttyyusb accessible for docker.

**Comment by coreyo** [comment]
@Yoda-x My previous posts should address your questions.  see: https://github.com/zigpy/bellows/issues/120#issuecomment-393627635

I don't even have docker installed on my system.

**Comment by Yoda-x** [comment]
Ok, I may have mixed it up with others. 
Other option is that the stick expects [other portspeed and/or flow control ](https://community.home-assistant.io/t/zigbee-stick-for-europe-any-available-retain-state-in-hass/17361/23?u=yoda-x)

**Comment by kenthua** [comment]
@Yoda-x at least within the docker image I've tried different baud rates (through bellows CLI flag), 9600,57600,115200 to no avail.  

I read a while ago that folks had the husbzb-1 working on rpi's, but eventually switched to other platforms because of lag, too many devices, other issues.  So it has worked at one point.  So possibly drivers, libs, code, changed at some point that affected this particular usb device?

**Comment by kenthua** [comment]
@pumbarger see my original comment https://github.com/zigpy/bellows/issues/120#issue-324668110, it's likely a red herring.

**Comment by coreyo** [comment]
For what it's worth, I tried plugging my dongle into one of my laptops running a more recent version of ubuntu (18.x I believe), creating a virtualenv, installing zigpy and bellows, then testing out bellows.  I only had python3.5.6, but I experienced the same issues.

**Comment by coreyo** [comment]
@kenthua no, I just got it about 2 weeks ago and it has never worked.  Is there a definitive way to test this in either windows or linux?

Given that we see so many people having this issue (including the related threads on other projects), I'm tempted to say that it is (in no particular order):
1. bellows
2. a large batch of defective devices
3. something in one of the kernel modules

**Comment by nphil** [comment]
I have the same issue, running in a docker container. 
`# bellows -v DEBUG -d /dev/ttyUSB1 -b 57600 info                                                                                                                                                                                                                                                                                                                                                                                     
debug: Using selector: EpollSelector                                                                                                                                                                                                                                                                                                                                                                                                 
debug: Connected. Resetting.                                                                                                                                                                                                                                                                                                                                                                                                         
debug: Sending: b'1ac038bc7e'`

Stuck there. I just got the stick 2 days ago and tried setting it up with Openhab and the Zigbee portion doesn't work there either. Beginning to think maybe we have an incompatible/newer zigbee firmware on the sticks?

**Comment by thehomerepot** [comment]
I just got this device and tested with both Hass.io and and docker and I get the same output from Bellows.

**Comment by egorian** [comment]
I'm also having the issue that bellows (Rpi + HomeAssistant inside a docker container) stuck after some time with the same message 

```
 $ docker exec -it hass0.68.1 bash
bash-4.4# bellows -v DEBUG -d /dev/zigbee info
debug: Using selector: EpollSelector
debug: Connected. Resetting.
debug: Sending: b'1ac038bc7e'
^C
Aborted!
```

Trying to isolate the issue, I disabled zha component and running the docker container.  And then trying to use bellows inside the container.  After more than 24 hours it seems that it bellows it's still working fine and responding. 

```
$ docker exec -it hass0.68.1 bash
bash-4.4# bellows -v DEBUG -d /dev/zigbee info
debug: Using selector: EpollSelector
debug: Connected. Resetting.
debug: Sending: b'1ac038bc7e'
debug: RSTACK Version: 2 Reason: RESET_SOFTWARE frame: b'c1020b0a527e'
debug: Send command version
debug: Sending: b'004221a850ed2c7e'
debug: Data frame: b'0142a1a8502805e67f627e'
debug: Sending: b'8160597e'
debug: Application frame 0 (version) received
debug: Configuring...
debug: Send command setConfigurationValue
debug: Sending: b'7d314321fb582815c3277e'
debug: Data frame: b'1243a1fb54fb737e'
debug: Sending: b'82503a7e'
debug: Application frame 83 (setConfigurationValue) received
debug: Send command setConfigurationValue
debug: Sending: b'224021fb592f15226f7e'
debug: Data frame: b'2340a1fb54c6107e'
debug: Sending: b'83401b7e'
debug: Application frame 83 (setConfigurationValue) received
debug: Send command setConfigurationValue
debug: Sending: b'334121fb792b15a2d77e'
debug: Data frame: b'3441a1fb54d32a7e'
debug: Sending: b'8430fc7e'
debug: Application frame 83 (setConfigurationValue) received
debug: Send command setConfigurationValue
debug: Sending: b'444621fb55d51534da7e'
debug: Data frame: b'4546a1fb5435d07e'
debug: Sending: b'8520dd7e'
debug: Application frame 83 (setConfigurationValue) received
debug: Send command networkInit
debug: Sending: b'554721bfb6607e'
debug: Data frame: b'5647a5bf54b4247e'
debug: Sending: b'8610be7e'
debug: Application frame 23 (networkInit) received
debug: Data frame: b'6647b1b1c487df7e'
debug: Sending: b'87009f7e'
debug: Application frame 25 (stackStatusHandler) received
debug: Send command getEui64
debug: Sending: b'6744218e08c37e'
debug: Data frame: b'7744a18eab0f06b959fb4725c7567e'
debug: Sending: b'8070787e'
debug: Application frame 38 (getEui64) received
[00:0d:6f:00:0b:13:25:ff]
debug: Send command getNodeId
debug: Sending: b'7045218f65587e'
debug: Data frame: b'0045a18f542af0d57e'
debug: Sending: b'8160597e'
debug: Application frame 39 (getNodeId) received
[0]
debug: Send command networkState
debug: Sending: b'014a21b0ba147e'
debug: Data frame: b'114aa1b05617a27e'
debug: Sending: b'82503a7e'
debug: Application frame 24 (networkState) received
[<EmberNetworkStatus.JOINED_NETWORK: 2>]
debug: Send command getNetworkParameters
debug: Sending: b'124b21803b0c7e'
debug: Data frame: b'224ba180542baec2c743095816f7264f944127abedce677302c15e3c7e'
debug: Sending: b'83401b7e'
debug: Application frame 40 (getNetworkParameters) received
[<EmberStatus.SUCCESS: 0>, <EmberNodeType.COORDINATOR: 1>, <EmberNetworkParameters extendedPanId=[187, 112, 158, 215, 67, 125, 188, 162] panId=1716 radioTxPower=8 radioChannel=15 joinMethod=EmberJoinMethod.USE_MAC_ASSOCIATION nwkManagerId=0 nwkUpdateId=0 channels=134215680>]
debug: Send command getCurrentSecurityState
debug: Sending: b'234821c160e47e'
debug: Data frame: b'3348a1c1545e154d7c874125c558921f717e'
debug: Sending: b'8430fc7e'
debug: Application frame 105 (getCurrentSecurityState) received
[<EmberStatus.SUCCESS: 0>, <EmberCurrentSecurityState bitmask=116 trustCenterLongAddress=00:0d:6f:00:0b:13:25:ff>]
```

I'm trying to figure it out that if any of the following reasons are the culprit:  
- Issues from HomeAssistant (Docker version) when setting up ZHA-bellows + stick.
- Timing issues between bellows and the stick  because of HomeAssistant (Docker version) that made the stick to reach a fail-state as @Yoda-x pointed out. (https://github.com/zigpy/bellows/issues/120#issuecomment-395151217). 
- Performance issues (I noticed that latest versions of HomeAssistant are more CPU/Memory intensive) that possibly make bellows to be slow in the communication between it and the stick (or make it not respond in the timely manner that the stick is expecting).

**Comment by kenthua** [comment]
@egorian That helps at least provide similar context to my debugging, which I noted that somehow it's related to the way the zha component is using the bellows and zigpy libraries.  

Unfortunately it still doesn't shed light on those who can't even get bellows standalone working.

@nphil I wonder if there was a way we could check firmware, that would help isolate the working/non-working instances if this does end up being the case.

**Comment by egorian** [comment]
> @egorian That helps at least provide similar context to my debugging, which I noted that somehow it's related to the way the zha component is using the bellows and zigpy libraries.
> 
> Unfortunately it still doesn't shed light on those who can't even get bellows standalone working.

After my tests I found that bellows needs exclusive access to the device, thus I disabled the zha component to be able to use bellows (kinda, standalone) directly from the HomeAssistant container.  Otherwise, once HomeAssistant fully started it's not possible to issue any command to bellows (maybe because HomeAssistant made the stick to reach a fail state).    

For those who want to replicate the tests, I recommend to stop the HomeAssistant docker container, rmmod cp210x (stick driver), then modprobe cp210x, verify that ttyUSB0/ttyUSB1 loaded correctly, disable zha component on HomeAssistant config file, start docker container and then try bellows cli from inside the container.

**Comment by egorian** [comment]
@thehomerepot did you try to reboot the device? unplug/plug the stick?   if the /dev/ttyUSB is correctly assigned, it's probably a hardware issue.  I can reproduce this behaviour using the **wrong** device (in my case, trying to use bellows with the zwave device of the stick as follow;  

```
bash-4.4# bellows -v DEBUG -d /dev/zwave info
debug: Using selector: EpollSelector
debug: Connected. Resetting.
debug: Sending: b'1ac038bc7e'
^C
Aborted!

```

That could means that bellows hangs or keeps waiting (indefinitely) for a response coming from the stick.

**Comment by thehomerepot** [comment]
@egorian Yes, I've tried rebooting my server. That didn't help. I'm working remotely so I can't unplug/plug it in.

**Comment by egorian** [comment]
@thehomerepot  I guess it worth to try to unplug/plug again (I'm not sure if the fail state could prevail to driver reloads/reboots).

Also, I'm pretty sure that you already verified the correct assignment of /dev/tty, But I will post this command for any other person that could reach this conversation to point them in the same direction that we currently are.

`sudo udevadm info -q property -a /dev/ttyUSB1`

**Comment by thehomerepot** [comment]
@egorian Yes, I validated /dev/ttyUSB1 was the correct device. I updated my comment with that syntax. I'll try unplug/plugging it in when I can.

**Comment by coreyo** [comment]
@egorian as demonstrated in my earlier posts, I'm in the same boat as @thehomerepot . I've tried unplugging/replugging/rebooting.  I'm running from a virtualenv.  I disabled zha in homeassitant a long time ago because it won't even start otherwise.  Bellows always hangs.  Is there possibly some windows program that we can use to test the device to rule out a defective device?

**Comment by kenthua** [comment]
@nphil This may be the device revision, don't think it's firmware version, but it's a start.

```
lsusb -d 10c4:8a2a -v | grep bcdDevice
Couldn't open device, some information will be missing
  bcdDevice            1.00
```

**Comment by thehomerepot** [comment]
@coreyo @kenthua @nphil 
On windows, I downloaded and installed [Axial Server](http://www.axialcontrol.com/download/downloadlist). I installed the unsigned drivers from this path "C:\Program Files (x86)\AxialControl.com\Axial Server\drivers\hubz", then launched [Simplicity Studio](https://www.silabs.com/products/development-tools/software/simplicity-studio).

Within this tool, it detected my device as "HubZ Smart Home Controller". I selected that controller name and hit the device configuration button. I selected part "CP2105-F01-GM" and board "CP2105 USB to Dual UART Bridge Evaluation Board" and hit ok. I then selected Run -> Connect to -> Silicon Labs ARM Part. A dialog box popped up and I selected "HubZ Smart Home Controller" and hit ok. I got a "problem occurred message, but closed it. The following was listed in the console output.

```
C:\SiliconLabs\SimplicityStudio\v4\developer\adapter_packs\usb\usb_monitor.exe C:\SiliconLabs\SimplicityStudio\v4\developer\adapter_packs\usb\usb_monitor.exe -d -s
C:\SiliconLabs\SimplicityStudio\v4\developer\adapter_packs\inspect_usbxpress\inspect_usbxpress.exe C:\SiliconLabs\SimplicityStudio\v4\developer\adapter_packs\inspect_usbxpress\inspect_usbxpress.exe -slist
deviceCount = 1
device (0) {
  SoftIndex = 0
  adapterLabel = HubZ Smart Home Controller (ID:0)
  SerialNo = 512006B3
  PartNumber = 5
  FirmwareVersion = 255
  Name = cp2105 version 255
  Type = CP210x
  Family = USBXpress
  Locked = 0
}
list: 04d9:a0cd@006:001[6&1B9AFB93&0&1] 046d:c247@007:005[6&65BD6FB&0&5] 046d:c247@006:002[6&1B9AFB93&0&2] 046d:c30a@008:003[5&10DC9A0C&0&3] 10c4:8a2a@009:001[512006B3] 
```

**Comment by nphil** [comment]
@kenthua @thehomerepot 

`[~] # lsusb -d 10c4:8a2a -v | grep bcdDevice
  bcdDevice            1.00`

Same on my device. Not sure if it means anything. 

I did the same thing, openhab refused to initialize the zigbee portion of the stick. Did you guys all purchase the stick from Amazon from seller Home Controls, Inc.  in the last 2-3 months? I've seen multiple posts on reddit by people with the same stick and everything works fine for them, but they're all older posts. We could try E-mailing the seller to see what's up, they have good ratings and exclusively sell home automation equipment.

**Comment by thehomerepot** [comment]
@nphil I did order from home Controls and did contact the seller. I got this message to download specific drivers which wouldn't even work on windows. I did not try calling them.

```
Hello *******,

Thank you for your order with Home Controls. The manufacturer has acknowledges the issue, and has provided the resolution below:

Nortek Tech Support has instructed users to go to Silicon Labs and get the USB to UART driver for the Silicon Labs EM3581 Zigbee Module.

Here's the link to the page where you can get the driver. It's a zip file and cannot be emailed. It has to downloaded.
Extract the drivers and load them into the USB stick.
Choose the EXTRX2USB and ETRX3USB Driver version 6.6.1

https://www.silabs.com/support/resources.ct-software.p-wireless_zigbee-and-thread.p-wireless_zigbee-and-thread_em35x 

Should you have any questions, please feel free to contact our Customer Service Team at 800-266-8765, Mon-Fri, 7am-5pm PST.

Thank you!

Best,

Customer Service
Home Controls 
```

**Comment by kenthua** [comment]
@nphil yea, I ordered mine late April on Amazon sold by Home Controls.  So there may be bad batches or different revisions from other locations.  Mine works as long as I'm not in a docker container.

@thehomerepot I wonder if you would have the same issues on Linux.  Maybe the Linux module/driver hasn't been updated to handle whatever has changed.  The windows driver changes could very well just be specific to the windows driver because of bugs in a previous version.

**Comment by nphil** [comment]
@thehomerepot 

Just sent them an E-mail too. Good to know that it's at least a known issue. 

@kenthua Mine does not work either in our out of a docker container :( Maybe I'm doing something wrong?

Also on the drivers page [https://www.silabs.com/support/resources.ct-software.p-wireless_zigbee-and-thread.p-wireless_zigbee-and-thread_em35x](url) there are some ETRX2USB drivers for linux 3.x, wonder if using these drivers instead of the default ones that come with whatever linux distro would help?

I'm running a 4.x kernel, so I'm not sure if the 3.x cp210x drivers would work.

**Comment by thehomerepot** [comment]
@kenthua I was originally testing on Linux. I only tested on a windows machine to confirm the issue wasn't specific to the kernel/drivers. The driver for windows has timestamps from 2013, so I don't think anything changed there.

**Comment by kenthua** [comment]
@thehomerepot There is one for 4.x https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers 

I don't think the GPIO pins are used for what we need and aren't in main kernel because it wasn't developed according to standards.

I checked the kernel source and it already has the SET_BAUDRATE/GET_BAUDRATE
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/usb/serial/cp210x.c

**Comment by nphil** [comment]
Well I opened up the stick and found both Z-wave and Zigbee soc's present, so at least I have the right stick. Do you guys think it's an issue with the cp2105 USB to UART bridge or a problem with the zigbee module firmware? I saw some notes on the EM3581 that the zigbee (ember something?) has been updated, not sure if this would affect anything.

I'm also trying to [talk to the zigbee modem with AT commands](https://www.silabs.com/documents/public/user-guides/TG-UG-0501-ETRX3CIEVK-EvaluationKitsUserGuide.pdf) via a serial terminal, but no go.

**Comment by egorian** [comment]
@nphil I don't think that the stick will respond to AT Commands.  Anyway, I also tried yesterday  (knowing that it will not work) to use minicom with no success.  
To summarize, I guess that we're dealing with different issues (seems that at least 3):
- Stick works fine for a limited period of time and then hangs using bellows + HomeAssistant (+ docker ?)
`debug: Sending: b'1ac038bc7e'`
- Stick doesn't responds at all also without using HomeAssistant and HomeAssistant doesn't start when enabled ZHA component
`debug: Sending: b'1ac038bc7e'`
- Stick got an error when using HomeAssistant (+ virtualenv ?)
` Error frame: b'c20251a8bd7e'.`

**Comment by coreyo** [comment]
I go mine here:  https://www.ebay.com/itm/122055230717 .   The price goes up and down, mine was much cheaper than what I see now.  Anyway, definitely the same problem from a different vendor.

**Comment by nphil** [comment]
anyone got any ideas or made any progress with this? I'm about to return the stick back to Amazon. What do you guys recommend for a cheap Z-wave/Zigbee combo that works well with HASS?

**Comment by kenthua** [comment]
@nphil While mine does work, there are definitely some limitations with the zha/bellows implementation.  One being the issue of not working, but others once it does work could be:

- lack of compatibility with your devices
- not reporting battery for devices which can lead to issues depending on the use of the devices

As far as future investment goes, I'd be looking to zwave devices.

**Comment by nphil** [comment]
Thanks, I have a Smartthings hub already and have all of my zigbee devices paired to it and talking to HASS via a MQTT bridge. Looks like I'll just stick with it until there's a more reliable long-term solution for zigbee.   I'm definitely sticking to Z-wave for all my future sensor purchases. On the other hand, cheap xiaomi sensors are so difficult to give up!

**Comment by kenthua** [comment]
I was on the path of moving from ST to pure HA, but I'm hesitant to move all my devices over now.  @Yoda-x and his fork of zha does do battery I believe for the xiaomi sensors.

**Comment by egorian** [comment]
Strangely enough, I flashed Hass.io on a different SD Card and put it on a different raspberryPI (Rpi3B) but using the same zwave+zigbee stick and then I moved all my zigbee devices to this environment.   After 24 hours is still working and not longer presenting this issue:
```
Stick works fine for a limited period of time and then hangs using bellows + HomeAssistant (+ docker ?)
debug: Sending: b'1ac038bc7e'
```

I'm guessing that my previous environment ((Rpi3B with Raspbian+docker) have some kind of resource (CPU? memory? USB timing?) issues that somehow made that bellows doesn't respond in a timely manner to the stick and the stick fall down on a failure state.

**Comment by iRanduMi** [comment]
Just found this thread and wanted to mention that I'm attempting to configure this same stick on an UNRAID docker instance of Home Assistant and have been experiencing the exact same issues.  Not really sure what else to do...

**Comment by nphil** [comment]
I managed to talk to someone at Home Controls about this and they're actually working with Nortek with this. He mentioned something about Nortek not flashing the Zigbee firmware properly, and how they sold a lot of sticks to Duke energy and they also had the same issue as us.

Regardless, he said there were some replacement sticks with proper Zigbee firmware loaded coming in around next month, so I'd recommend that you guys give Home Controls (or whoever your reseller was) a call - they assured me a free replacement.

**Comment by kenthua** [comment]
That's great news for those who can't get it working at all.

It doesn't fix it where it's broken in a docker container.

**Comment by iRanduMi** [comment]
Thank you for the follow up nphil!  I've contacted Home Controls, Inc.

**Comment by steve28** [comment]
FWIW, I have a HUSBZB-1 and have been using it with HA in docker on intel NUC since ~Jan.  A couple of weeks ago, all of my zigbee stuff (about 5 devices) stopped working. HA would load however, but I would get the warning "Setup is taking longer than 10s" in the log.  

Today I finally set about tackling this.  Updated HA, updated Ubuntu, restarted server, etc.  Not until I unplugged/re-plugged the stick and THEN rebooted, would it come back.

**Comment by kenthua** [comment]
It worked a few times in a docker image in the beginning.  After that no reboots, clean installs, zibgee.db wipes could bring it back.  After switching to native virtualenv it's been fairly stable.

**Comment by egorian** [comment]
After 5 days of being trying hass.io, the stick is still working fine. Now, I'll not been able to perform more test for the following days, but after those days I'll try to put the hass.io sdcard on the "old" rpi3 (where I was using Raspbian+Docker) to see if hass.io did the trick or if there is any other factor involved, eg.  Power supply, case (the "new" rpi3 is using a flirc case that dissipates heat better than the official case that I'm using on the "old" one), or other hardware issue.  I'll post the results here once I be able to perform those tests.

**Comment by iRanduMi** [comment]
Just want to follow up on this.  As previously mentioned, I had ordered my HUSBZB-1 stick via Home Controls, Inc.  After contacting them, they informed me that they've had quite a few of these issues with people.  I ordered a new one through them and am returning my current one via Amazon.  I received the new stick yesterday and bam, worked instantly.  The original was definitely a defective unit.

**Comment by nphil** [comment]
Just got my replacement from Homecontrols also - can confirm that the issue is fixed. Zigbee and Z-wave interfaces are detected. Now to figure out just how the hell I'm supposed to pair a Zigbee device, the process seems a bit more involved than my old ST hub..

**Comment by iRanduMi** [comment]
It's actually really easy.  Go to Services > zha.permit.  Call the service.  Go to your light and turn it off an on.  It'll likely appear within HA.  If not, you may need to reset the light if it's still attached to ST.  

My main issue with Zigbee support is that HA doesn't poll/monitor the status of the light.  So if you turn off the lamp when the light was previously on, HA still thinks it's on.  Same with rebooting HA or anything like that.

**Comment by BobAmelung** [comment]
I received my replacement HUSBZB-1 stick from Home Controls yesterday.  Same issue :-(

I enable zha, the web UI is inaccessible and I see "'utf-8' codec can't decode byte 0x80 in position 99: invalid start byte" in the zigbee.db

**Comment by pumbarger** [comment]
After reading someone finally has success after returning and replaced the HUSBZB-1 stick I decided to do the same.  I figured the safest bet would be to purchase one from the same vendor as the person who had success. So I purchased one from Home Controls via Amazon.

I wanted so start from scratch so I removed all z-wave devices from current controller and flashed a new hass.io image.  After first boot I added samba so I could edit the configuration.yaml file and add the basic config for z-wave and zha.

I restarted and it did not hang as it always had before, configuring zha. When brought up the web interface the z-wave controller was there, as before.  I then checked services and found zhz.permit and zha.remove.

I have not tried to add any devices yet because while waiting to see if this problem would resolve, I replaced the Zigbee bulbs I had with Z-wave bulbs which I prefer because they also act as repeaters.

So it appears the problem is a batch of bad devices which I would venture a guess that it is bad firmware rather than hardware.

It would be nice to know if there is a way to identify the good/working sticks from the bad.

Thanks to all you came up with theories on the source of the problem and who tested a myriad of things to prove or disprove the theories.

**Comment by matt1** [comment]
I am not so sure it is just batch of "bad devices".

I have a cp210x based zigbee USB stick which is not a HUSBZB-1.  It works 100% fine in Windows 10, and on a RPi3 with OpenHAB - zigbee devices are detected and can be controlled no worries so the firmware on the stick is fine.  However it does not work at all with Hass.io on the same RPi3, and in fact causes hass.io to not even start as noted in this and other issues.

tl;dr - Please dont close this resolved as there is a larger issue here I think.

**Comment by kenthua** [comment]
This thread morphed into a firmware / "bad device" thread.  I originally started it because there is an inherent initialization issue with bellows/zigpy and this device in a docker container.  Inconsistent behavior to not properly initializing at all in a container.  Since it appears a majority are not running in a container, it's not being looked into.  I have since given up on this device and HA handling my zigbee/zwave devices.  One crucial missing element is lack of battery sensor, which is a fairly significant problem.

**Comment by ajayjohn** [comment]
I agree. I got a HUSBZB-1 replacement as well and that has not fixed anything.
I hope the following logs are helpful. These are from my hassio setup on an Intel NUC running Proxmox on Debian.

**Home Assistant (won't startup unless zha is commented)**
`2018-10-12 11:52:22 WARNING (MainThread) [homeassistant.setup] Setup of zha is taking over 10 seconds.
2018-10-12 11:52:22 ERROR (MainThread) [bellows.ezsp] Exception running handler
asyncio.base_futures.InvalidStateError: invalid state
2018-10-12 11:52:22 ERROR (MainThread) [bellows.ezsp] Exception running handler
asyncio.base_futures.InvalidStateError: invalid state
2018-10-12 11:52:22 ERROR (MainThread) [bellows.ezsp] Exception running handler
asyncio.base_futures.InvalidStateError: invalid state
2018-10-12 11:52:22 DEBUG (MainThread) [bellows.uart] Error frame: b'c20251a8bd7e'
2018-10-12 11:52:22 DEBUG (MainThread) [bellows.uart] Error frame: b'c20251a8bd7e'
2018-10-12 11:52:22 DEBUG (MainThread) [bellows.uart] Error frame: b'c20251a8bd7e'
2018-10-12 11:52:22 DEBUG (MainThread) [bellows.uart] Error frame: b'c20251a8bd7e'
2018-10-12 11:52:22 DEBUG (MainThread) [bellows.uart] Error frame: b'c20251a8bd7e'
2018-10-12 11:52:22 DEBUG (MainThread) [bellows.uart] Error frame: b'c20251a8bd7e'
2018-10-12 11:52:22 DEBUG (MainThread) [bellows.uart] Error frame: b'c20251a8bd7e'
2018-10-12 11:52:22 DEBUG (MainThread) [bellows.uart] Error frame: b'c20251a8bd7e'
2018-10-12 11:52:22 DEBUG (MainThread) [bellows.uart] Error frame: b'c20251a8bd7e'
2018-10-12 11:52:22 DEBUG (MainThread) [bellows.uart] Error frame: b'c20251a8bd7e'`

** dmesg on host startup **
`[   46.173972] usbserial: USB Serial support registered for cp210x
[   46.174001] cp210x 2-2:1.0: cp210x converter detected
[   46.177116] usb 2-2: cp210x converter now attached to ttyUSB0
[   46.177137] cp210x 2-2:1.1: cp210x converter detected
[   46.178395] usb 2-2: cp210x converter now attached to ttyUSB1
[   46.189659] cp210x ttyUSB1: failed set req 0x1e size 4 status: -32
[   46.189662] cp210x ttyUSB1: failed to set baud rate to 300`

**Comment by drewzh** [comment]
For what it's worth, I've been attempting to get Zigbee to work, on and off, for over a week now. I've tried using an Elelabs USB and a Telegenesis USB device (thinking the problem was due to the hardware). This is using both Docker on a Synology NAS and also using hass.io on a raspberry pi 3 (another avenue I went down to try and resolve. Same results as users above. Note that I'm using the very latest release of home assistant 0.81.1

**Comment by jooprzol** [comment]
> 
> 
> FWIW, I have a HUSBZB-1 and have been using it with HA in docker on intel NUC since ~Jan. A couple of weeks ago, all of my zigbee stuff (about 5 devices) stopped working. HA would load however, but I would get the warning "Setup is taking longer than 10s" in the log.
> 
> Today I finally set about tackling this. Updated HA, updated Ubuntu, restarted server, etc. Not until I unplugged/re-plugged the stick and THEN rebooted, would it come back.

This did it for me. Though, in my case, I shut down, unplugged the stick and then powered up the machine and plugged it back in as it was booting. HASS sees the device and brings up the web UI now, whereas before it would hang and never load the UI.

**Comment by drewzh** [comment]
Did anybody make any progress on this? I switched back to SmartThings for 6 months and came back to see if I could finally get it to work... alas, same issue again :(

**Comment by matt1** [comment]
I think the conclusion I came to is that this binding is for a very specific chipset and does not support most devices.  I've given up on Home Assistant as a result - I dont see how it can ever work unless you have the exact same precise hardware the author has.

**Comment by drewzh** [comment]
I've now landed on a smartthings and HA mix, with smartthings doing the radio and HA doing the logic. I wish I could easily get rid of the smartthings element though. ZWave is fine, ZigBee is a no go. I've tried about everything I possibly could with multiple USB sticks and different hosts (docker Vs raspberry pi etc). No dice.

**Comment by dmulcahey** [comment]
When is the last time either of you tried using ZHA? just curious?

**Comment by matt1** [comment]
Tried again today with fresh installation of home assistant and a new dongle.  Same problem as before.

**Comment by Adminiuga** [comment]
~~you are doing it wrong~~it doesn't work. Don't try it again

**Comment by drewzh** [comment]
@dmulcahey about 2 weeks ago. I believe it was 0.90.

**Comment by Adminiuga** [comment]
@drewzh but what is exactly is "your" issue? Earlier you posted:
>  This is using both Docker on a Synology NAS and also using hass.io on a raspberry pi 3 (another avenue I went down to try and resolve. Same results as users above.

and users above have at least two types of issues:
1. stick is not responding at all
2. zigbee starts, but some time later stops working

The two above have different root causes. The 2nd one occurs most often on Pi hardware and should be mitigated in 0.91.x
The 1st one, assuming working hardware is most likely related to local setup.

**Comment by dmulcahey** [comment]
@matt1 this is incorrect. Xbee, ember and deconz radios are all supported now.

**Comment by matt1** [comment]
> @matt1 this is incorrect. Xbee, ember and deconz radios are all supported now.

Thats fine - I guess those are all EZSP compatible?  It seems like this library _only_ talks EZSP and not other protocols.  Not every dongle is not EZSP-compatible though (my telegesis dongle wasn't for example, but worked fine in openHab whose driver supported it) and so this is never going to work for non-EZSP dongles.  No amount of "setup" work is ever going to make non-EZSP dongles work from what I can tell.

I wish I'd realised this a lot earlier as it would have saved a _lot_ of time fiddling about! :-)  Hopefully this saves some other people's sanity in this area! :-)

**Comment by Adminiuga** [comment]
@dmulcahey 
> Unix is user-friendly. It's just very selective about who its friends are.

I guess the same applies to ZHA...

**Comment by dmulcahey** [comment]
@matt1 you may want to reread my previous reply and maybe search the other 2 radios I mentioned before you reply further... For anyone who stumbles across this in the future: Xbee and Deconz are not ember (EZSP) radios.

**Comment by Adminiuga** [comment]
ah, that's explains why there're [zigpy-deconz](https://github.com/zigpy/zigpy-deconz) and [zigpy-xbee](https://github.com/zigpy/zigpy-xbee)

**Comment by matt1** [comment]
> @matt1 you may want to reread my previous reply and maybe search the other 2 radios I mentioned before you reply further... For anyone who stumbles across this in the future: Xbee and Deconz are not ember (EZSP) radios.

Ok you may want to update your README.md then because it says this and is quite misleading (and is why I was confused):

> `bellows` is a Python 3 project to implement support for EmberZNet devices using the EZSP protocol.

**Comment by thehomerepot** [comment]
> Ok you may want to update your README.md then because it says this and is quite misleading (and is why I was confused):

This repo is specifically for implementing EZSP devices with bellows, not for HA as a whole. Look at the links @Adminiuga posted and see how there are separate repos for other devices as well.

**Comment by matt1** [comment]
Yeah I see that now - looks like home assistant defaults to this library as its zigbee support though.

**Comment by Adminiuga** [comment]
Closing on no activity.

**Comment by zombielinux** [comment]
Problem is still present for me.

```
bellows-homeassistant    0.9.0    
homeassistant            0.96.5   
zigpy                    0.2.0    
zigpy-deconz             0.2.1    
zigpy-homeassistant      0.7.0    
zigpy-xbee-homeassistant 0.4.0  
```
```
[root@changeme /]# sudo bellows -v DEBUG -d /dev/ttyUSB1 info
debug: Using selector: EpollSelector
debug: Using selector: EpollSelector
debug: Connected. Resetting.
debug: Resetting EZSP
debug: Resetting ASH
debug: Sending: b'1ac038bc7e'
Traceback (most recent call last):
  File "/usr/bin/bellows", line 10, in <module>
    sys.exit(main())
  File "/usr/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python3.7/site-packages/bellows/cli/util.py", line 40, in inner
    loop.run_until_complete(f(*args, **kwargs))
  File "/usr/lib/python3.7/asyncio/base_events.py", line 584, in run_until_complete
    return future.result()
  File "/usr/lib/python3.7/site-packages/bellows/cli/ncp.py", line 75, in info
    s = await util.setup(ctx.obj['device'], ctx.obj['baudrate'])
  File "/usr/lib/python3.7/site-packages/bellows/cli/util.py", line 107, in setup
    await s.reset()
  File "/usr/lib/python3.7/site-packages/bellows/ezsp.py", line 55, in reset
    await self._gw.reset()
  File "/usr/lib/python3.7/site-packages/bellows/uart.py", line 208, in reset
    return await asyncio.wait_for(self._reset_future, timeout=RESET_TIMEOUT)
  File "/usr/lib/python3.7/asyncio/tasks.py", line 423, in wait_for
    raise futures.TimeoutError()
concurrent.futures._base.TimeoutError
^CError in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python3.7/concurrent/futures/thread.py", line 40, in _python_exit
    t.join()
  File "/usr/lib/python3.7/threading.py", line 1032, in join
    self._wait_for_tstate_lock()
  File "/usr/lib/python3.7/threading.py", line 1048, in _wait_for_tstate_lock
    elif lock.acquire(block, timeout):
KeyboardInterrupt
```
This is on an archlinux vm running on Proxmox6 with USB passthrough. The exact same result occurs if I use direct serial pass through, i.e.
```
qm  set 119 -serial1 /dev/ttyUSB1
```

This error also occurs on the Proxmox6 hypervisor running nothing at all as verified by 
```
lsof | grep /dev/ttyUSB1
```
